content/en/altinity-kb-queries-and-syntax/trace_log.md

```diff
@@ -37,56 +37,78 @@ pip3 install -r requirements.txt
 ```
 
 For debugging particular query:
-```
+1. In the clickhouse-client, set your desired profiler settings:
+```sql
 clickhouse-client 
 
 SET query_profiler_cpu_time_period_ns=1000000; -- 1000 times per 'cpu' sec
--- or SET query_profiler_real_time_period_ns=2000000; -- 500 times per 'real' sec.
--- or SET memory_profiler_sample_probability=0.1; -- to debug the memory allocations
-
+-- or
+SET query_profiler_real_time_period_ns=2000000; -- 500 times per 'real' sec.
+-- or
+SET memory_profiler_sample_probability=0.1; -- to debug the memory allocations
+```
+2. Execute your query:
+```sql
 SELECT ... <your select>
-
+```
+3. Flush the logs:
+```sql
 SYSTEM FLUSH LOGS;
-
--- get the query_id from the clickhouse-client output or from system.query_log (also pay attention on query_id vs initial_query_id for distributed queries).
 ```
+4. Retrieve the query_id from either the clickhouse-client output or from system.query_log (note the difference between query_id and initial_query_id for distributed queries).
+5. Process the stack traces:
```

Comment:
There is now also a global profiler, and obe more setting for memory - collecting peaks, and step